src/modules/kvs/kvs.c

```diff
@@ -723,11 +723,16 @@ static void get_request_cb (flux_t *h, flux_msg_handler_t *mh,
 
     /* if bad lh, then first time rpc and not a replay */
     if (lookup_validate (arg) == false) {
+        const char *namespace;
         struct kvsroot *root;
 
         ctx = arg;
 
-        if (!(root = getroot (ctx))) {
+        if (flux_request_unpack (msg, NULL, "{ s:s }",
```

Comment:
It may tighten things up a bit if the namespace parameter is required instead of optional.